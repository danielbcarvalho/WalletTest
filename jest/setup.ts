import 'react-native-gesture-handler/jestSetup';
import { mockDefaultTheme } from '../src/theme/defaultTheme';

jest.mock('styled-components', () => {
  const original = jest.requireActual('styled-components');

  return {
    ...original,
    useTheme: () => {
      return mockDefaultTheme;
    },
  };
});
