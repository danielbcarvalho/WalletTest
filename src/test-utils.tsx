// TODO: Verificar essa melhoria nos testes
import React from 'react';
import {
  render,
  RenderAPI,
  RenderOptions,
} from '@testing-library/react-native';
import renderer, { ReactTestRenderer } from 'react-test-renderer';
import { ThemeProvider } from 'styled-components/native';

import { mockDefaultTheme } from './theme/defaultTheme';

interface IAllThemeProviders {
  children: React.ReactElement;
}
const AllTheProviders = ({ children }: IAllThemeProviders) => {
  return <ThemeProvider theme={mockDefaultTheme}>{children}</ThemeProvider>;
};

function customRender(
  ui: React.ReactElement,
  options = {} as RenderOptions,
): RenderAPI {
  return render(ui, { wrapper: AllTheProviders, ...options });
}

export function rendererCreate(
  component: React.ReactElement,
): ReactTestRenderer {
  return renderer.create(<AllTheProviders>{component}</AllTheProviders>);
}

// re-export everything
export * from '@testing-library/react-native';

// override render method
export { customRender as render };
