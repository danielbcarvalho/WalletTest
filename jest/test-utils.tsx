import React, { ReactElement, FC } from 'react';
import { render } from '@testing-library/react-native';
import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
import { ThemeProvider } from '../src/theme';

type Options = Parameters<typeof render>[1];

const queryClient = new QueryClient({
  defaultOptions: {
    queries: {
      cacheTime: Infinity,
    },
  },
});
const AllTheProviders: FC<{ children: React.ReactNode }> = ({ children }) => (
  <ThemeProvider>
    <QueryClientProvider client={queryClient}>{children}</QueryClientProvider>
  </ThemeProvider>
);

const customRender = (ui: ReactElement, options?: Options) =>
  render(ui, { wrapper: AllTheProviders, ...options });

export * from '@testing-library/react-native';
export { customRender as render };
