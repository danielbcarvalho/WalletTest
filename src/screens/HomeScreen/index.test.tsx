import React from 'react';
import { fireEvent } from '@testing-library/react-native';
import { useNavigation } from '@react-navigation/native';

import { render } from '../../../jest/test-utils';

import Home from './';

jest.mock('@react-navigation/native', () => ({
  useNavigation: jest.fn(),
}));

describe('Home', () => {
  let navigateMock: jest.Mock;

  beforeEach(() => {
    navigateMock = jest.fn();
    useNavigation.mockReturnValue({ navigate: navigateMock });
  });

  test('navigates to CardListScreen on my cards button press', () => {
    const { getByText } = render(<Home />);
    const myCardsButton = getByText('my cards');
    fireEvent.press(myCardsButton);

    expect(navigateMock).toHaveBeenCalledWith('WalletAnimatedScreen');
  });

  test('navigates to CardRegistrationScreen on register card button press', () => {
    const { getByText } = render(<Home />);
    const registerCardButton = getByText('register card');
    fireEvent.press(registerCardButton);

    expect(navigateMock).toHaveBeenCalledWith('CardRegistration');
  });

  test('matches snapshot', () => {
    const { toJSON } = render(<Home />);
    expect(toJSON()).toMatchSnapshot();
  });
});
