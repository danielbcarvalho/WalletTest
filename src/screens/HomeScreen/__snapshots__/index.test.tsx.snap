// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home matches snapshot 1`] = `
<RNCSafeAreaView
  edges={
    {
      "bottom": "additive",
      "left": "additive",
      "right": "additive",
      "top": "additive",
    }
  }
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#142995",
      "flexBasis": 0,
      "flexGrow": 1,
      "flexShrink": 1,
      "justifyContent": "center",
      "paddingBottom": 10,
      "paddingLeft": 30,
      "paddingRight": 30,
      "paddingTop": 10,
    }
  }
>
  <View
    style={
      {
        "backgroundColor": "#EEEEEE",
        "borderBottomLeftRadius": 50,
        "borderBottomRightRadius": 50,
        "height": 230,
        "opacity": 0.2,
        "position": "absolute",
        "right": 120,
        "top": -70,
        "transform": [
          {
            "rotate": "-39.93deg",
          },
        ],
        "width": 370,
      }
    }
    top={true}
  />
  <View
    style={
      {
        "backgroundColor": "#EEEEEE",
        "borderBottomLeftRadius": 50,
        "borderBottomRightRadius": 50,
        "bottom": -70,
        "height": 230,
        "left": 120,
        "opacity": 0.2,
        "position": "absolute",
        "transform": [
          {
            "rotate": "139deg",
          },
        ],
        "width": 370,
      }
    }
  />
  <Text
    style={
      {
        "color": "#FFF",
        "fontFamily": "PTSans-Caption",
        "fontSize": 28,
        "lineHeight": 52,
      }
    }
  >
    Wallet Test
  </Text>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": false,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "alignSelf": "stretch",
        "backgroundColor": "#12C2E9",
        "borderBottomLeftRadius": 12,
        "borderBottomRightRadius": 12,
        "borderTopLeftRadius": 12,
        "borderTopRightRadius": 12,
        "justifyContent": "center",
        "marginBottom": 10,
        "marginLeft": 0,
        "marginRight": 0,
        "marginTop": 10,
        "maxHeight": 56,
        "minHeight": 56,
        "opacity": 1,
        "paddingBottom": 10,
        "paddingLeft": 20,
        "paddingRight": 20,
        "paddingTop": 10,
      }
    }
    testID="multipurpose-button"
  >
    <Text
      style={
        {
          "color": "#FFFFFF",
          "fontFamily": "PTSans-Regular",
          "fontSize": 18,
        }
      }
      type="primary"
    >
      my cards
    </Text>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": false,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "alignSelf": "stretch",
        "backgroundColor": "#A5FF32",
        "borderBottomLeftRadius": 12,
        "borderBottomRightRadius": 12,
        "borderTopLeftRadius": 12,
        "borderTopRightRadius": 12,
        "justifyContent": "center",
        "marginBottom": 10,
        "marginLeft": 0,
        "marginRight": 0,
        "marginTop": 10,
        "maxHeight": 56,
        "minHeight": 56,
        "opacity": 1,
        "paddingBottom": 10,
        "paddingLeft": 20,
        "paddingRight": 20,
        "paddingTop": 10,
      }
    }
    testID="multipurpose-button"
  >
    <Text
      style={
        {
          "color": "#142995",
          "fontFamily": "PTSans-Regular",
          "fontSize": 18,
        }
      }
      type="secondary"
    >
      register card
    </Text>
  </View>
</RNCSafeAreaView>
`;
